### PR TITLE
Add sprite sequence documentation generation

### PR DESCRIPTION
**Pull request by penev92**
A continuation to #19948, #19986, #20087, #20250.

You can see the final result at  https://docs.openra.net/en/test/dev/sequences/

### CODE REVIEW

File: OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			Game.ModData = utility.ModData;
+
+			var version = utility.ModData.Manifest.Metadata.Version;
+			if (args.Length > 1)
+				version = args[1];
+
+			var objectCreator = utility.ModData.ObjectCreator;
+			var traitInfos = objectCreator.GetTypesImplementing<ISpriteSequence>().OrderBy(t => t.Namespace);
```
Comment (PunkPun) [inline OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs]: `
```suggestion
			var sequenceInfos = objectCreator.GetTypesImplementing<ISpriteSequence>().OrderBy(t => t.Namespace);
```

**Review by PunkPun — COMMENTED** [review]
LGTM. But just a questions, why is `Properties` code formatted completely different to weapons and traits? Is it because of lack of Attributes?

I assume github actions will be added in a follow up

**Comment by penev92** [comment]
I guess I somehow missed the GHAction update when I was switching branches and didn't even notice. The `traitInfos` is a good catch. The `Properties` code is formatted differently because here we precalculate some things that take up screen space and in the other two commands those were mostly one-liners.

File: OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+			var json = GenerateJson(version, traitInfos);
+			Console.WriteLine(json);
+		}
+
+		static string GenerateJson(string version, IEnumerable<Type> sequenceTypes)
+		{
+			var sequenceTypesInfo = sequenceTypes.Where(x => !x.ContainsGenericParameters && !x.IsAbstract)
+				.Where(x => x.Name != nameof(FileNotFoundSequence)) // HACK: This is the simplest way to exclude FileNotFoundSequence, which shouldn't be added.
```
Comment (Mailaender) [inline OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs]: Not really a hack though.

Comment (penev92) [reply OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs]: I'll change to "NOTE:" I guess. I think it is still worth noting.
It's also here to match the other two JSON exporting commands.

**Comment by penev92** [comment]
Fixed.

**Comment by penev92** [comment]
Fixed. Also added a link to the PR description with the test/dev version of the docs site where you can see the final result.

**Review by PunkPun — APPROVED** [review]

File: OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs (hunk 2 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/UtilityCommands/ExtractSequenceDocsCommand.cs]: The `FileNotFoundSequence` is the hack I guess.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6d02239bac5e519014d22a3ed2858139ebe4d898...728c69842865cec17fd962d9fc41b969f1e568c0)

**Pull request merged**